test/functional/eval/capture_spec.lua

```diff
@@ -0,0 +1,39 @@
+local helpers = require('test.functional.helpers')
+local eq = helpers.eq
+local eval = helpers.eval
+local clear = helpers.clear
+local execute = helpers.execute
+local source = helpers.source
+
+describe('capture()', function()
+  before_each(clear)
+
+  it('returns the same result with :redir', function()
+    execute('redir => g:foo | messages | redir END')
+    eq(eval('g:foo'), eval('capture(":messages")'))
```

Comment:
This is easier expressed with `eq(redir_exec('messages'), funcs.capture('messages'))` (both new variables live in helpers module).